cirq-core/cirq/devices/insertion_noise_model.py

```diff
@@ -63,6 +63,10 @@ def noisy_moment(
                 noise_ops.append(self.ops_added[match_id])
         if not noise_ops:
             return [moment]
+
+        from cirq import circuits
+
+        noise_steps = circuits.Circuit(noise_ops)
         if self.prepend:
-            return [ops.Moment(noise_ops), moment]
-        return [moment, ops.Moment(noise_ops)]
+            return [*noise_steps.moments, moment]
+        return [moment, *noise_steps.moments]
```

Comment:
This construction is necessary for handling noise which affects qubits other than those found in the operation (e.g. for crosstalk), as that may produce two noise ops on the same qubit. See tests for an example.

Reply:
Python guarantees that dictionaries iterate in insertion order, so if op_ids A and B both match a gate but neither is a subtype of the other, this will consistently match whichever one appears first in `ops_added`.

I've updated the class docstring to clarify this behavior.